src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -1480,9 +1480,9 @@ void Assembler::addl(Register dst, Register src) {
 
 void Assembler::eaddl(Register dst, Register src1, Register src2, bool no_flags) {
   InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
-  (void) evex_prefix_and_encode_ndd(src1->encoding(), dst->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  (void) evex_prefix_and_encode_ndd(src2->encoding(), dst->encoding(), src1->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
```

Comment:
Hi @vamsi-parasa,
NDD is very flexible in terms of argument selection, i.e. 
ADDL  NDD,  SRC1 (ModRM.R/M), SRC2 (ModRM.REG) has opcode 0x01
Whereas,
ADDL  NDD, SRC1 (ModRM.REG),  SRC2 (ModRM.R/M) has opcode 0x03

In this case, we are trying to match GCC encoding scheme.

Can you please add the following comment here since the argument nomenclature does not match with parameter nomenclature?

NDD shares its encoding bits with NDS bits for regular EVEX instruction. Therefore we are passing DST as the second argument to minimize changes in leaf level routine.

Reply:
Thanks for addressing this.

Reply:
> Hi @vamsi-parasa, NDD is very flexible in terms of argument selection, i.e. ADDL NDD, SRC1 (ModRM.R/M), SRC2 (ModRM.REG) has opcode 0x01 Whereas, ADDL NDD, SRC1 (ModRM.REG), SRC2 (ModRM.R/M) has opcode 0x03
> 
> In this case, we are trying to match GCC encoding scheme.
> 
> Can you please add the following comment here since the argument nomenclature does not match with parameter nomenclature?
> 
> NDD shares its encoding bits with NDS bits for regular EVEX instruction. Therefore we are passing DST as the second argument to minimize changes in leaf level routine.

Thanks for addressing this.